xla/backends/gpu/runtime/nccl_collective_permute_thunk.cc

```diff
@@ -237,12 +250,30 @@ absl::Status NcclCollectivePermuteStartThunk::RunNcclCollective(
                     p2p_memcpy_enabled_;
 
   TF_ASSIGN_OR_RETURN(GpuCollectives * collectives, GetGpuCollectives(params));
+  std::optional<int64_t> source_id = source_target.source;
+  std::optional<int64_t> target_id = source_target.target;
   if (use_memcpy) {
-    se::DeviceMemoryBase sync_var_address =
-        se::DeviceMemoryBase(barrier_flags_[current_id]->opaque());
-    TF_RETURN_IF_ERROR(comm_handle.comm->AllReduce(
-        sync_var_address, sync_var_address, PrimitiveType::U8, 1,
-        ReductionKind::MIN, GpuCollectives::On(stream)));
+    // For receiving side, record an event.
```

Comment:
The motivation is a bit unclear, so I think this warrants a more detailed comment. Here's an example, feel free to use or rewrite:

> When memcopying to another device, the source device must ensure the destination device's buffer is ready to receive the data. So have the receiving side record an event, then later have the sending side wait for the event.

Reply:
sg, added more explanation